File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/SinH.java
package uk.ac.uea.mathsthing.plugins.functions;

import java.math.BigDecimal;

import uk.ac.uea.mathsthing.IFunctionPlugin;
import uk.ac.uea.mathsthing.util.FormulaException;

public class SinH extends IFunctionPlugin {
	
	@Override
	public String getName()
	{
		return "sinh";
	}
	
	@Override
	public BigDecimal function(BigDecimal input) throws FormulaException
	{
		return new BigDecimal(Math.sinh(input.doubleValue()));
	}
}

File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/SinH.scala
package uk.ac.uea.mathsthing.plugins.functions

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin
import java.math.BigDecimal

class SinH extends IFunctionPlugin {

  def getName : String = "sinh"
    
  def function(input: BigDecimal) : BigDecimal = {
  	val result:BigDecimal = new BigDecimal(Math.sinh(input.doubleValue))
    result
  }
}